### ISSUE TITLE
Parallel CI: Rerunning 1 machine with failed tests reruns ALL spec files instead of only previously failed spec files assigned to that machine.

### ISSUE THREAD

**Issue by garyluu** [cypress-io/cypress#3995]
### Current behavior:
After running parallel tests on 3 machines/jobs on Travis CI; If for whatever reason we wanted to restart a Travis CI machine/job, it will re-run all the Cypress tests on that one Travis CI machine/job

### Desired behavior:

Restarting a single Travis CI machine/job would only re-run the Cypress tests that previous ran on that machine/job

### Versions

Cypress 3.2.0

**Comment by jennifer-shehane** [comment]
This shouldn't happen.

Can you provide more information about your Travis setup. A link to the Travis project would be most ideal (feel free to email support@cypress.io)

But, if you can provide the Travis yml file you run / how you run the cypress commands. Thanks!

**Comment by garyluu** [comment]
https://travis-ci.org/dockstore/dockstore-ui2/builds/523057315

Each machine took about 21 mins (as expected).  I restarted the 2nd job fairly soon after it failed and now at the time of writing it's surpassed 21 mins and expected to time out at 50 mins.

Relevant code snippets:
.travis.yml
```
jobs:
  include:
    - stage: integration test
      name: "Parallel Cypress Tests"
      script:
        - ./scripts/run-travis-script.sh

    - stage: integration test
      name: "Parallel Cypress Tests"
      script:
        - ./scripts/run-travis-script.sh

    - stage: integration test
      name: "Parallel Cypress Tests"
      script:
- ./scripts/run-travis-script.sh
```
where ./scripts/run-travis-script.sh is

```#!/bin/bash
set -o errexit
set -o pipefail
set -o nounset
set -o xtrace

npm run webservice
npm run build
npm run start &
./scripts/wait-for.sh
cypress run --record --config defaultCommandTimeout=10000 --parallel --group 3x-electron

```

**Comment by jennifer-shehane** [comment]
We've discussed this issue as a team and have come to the conclusion that it's not possible with our current architecture to trigger a rerun of the subset of spec files that ran on a specific machine. 

All that cypress sees on rerun is the `cypress run --record --parallel` command, so we do not know that your intention is to only run some spec files, or only run from a failed machine, or only run failed specs. 

There is somewhat of a workaround we've provided, to rerun failed specs upon the `cypress run` finishing. See the gist below. It requires use of the [Module API](https://on.cypress.io/module-api#cypress-run). 

https://gist.github.com/Bkucera/4ffd05f67034176a00518df251e19f58

We also have features in the works to improve this failure workflow. Like: 
- retrying failures automatically - https://github.com/cypress-io/cypress/issues/1313
- aborting the tests on first failure - https://github.com/cypress-io/cypress/issues/518
